aiida_quantumespresso/parsers/cp.py

```diff
@@ -39,140 +40,181 @@ def parse(self, **kwargs):
-            # TODO: Add an error for this counter
-            return self.exit(self.exit_codes.ERROR_MISSING_XML_FILE)
+
+        print_counter_xml=True
+        no_trajectory_output=False
+        if self.node.process_class._FILE_XML_PRINT_COUNTER_BASENAME not in list_of_files and self.node.process_class._FILE_PRINT_COUNTER_BASENAME not in list_of_files:
+            self.logger.error('We could not find the print counter file in the output (' + self.node.process_class._FILE_PRINT_COUNTER_BASENAME + ' or ' + self.node.process_class._FILE_XML_PRINT_COUNTER_BASENAME + ' not in {} )'.format(list_of_files) + 'assuming no trajectory output was produced' )
+            no_trajectory_output=True
+            #this can happen and it is not an error!
+        if not no_trajectory_output:
+            if self.node.process_class._FILE_PRINT_COUNTER_BASENAME in list_of_files:
+                self.logger.info('print counter not in xml format')
+                print_counter_xml=False
+                FILE_PRINT_COUNTER_BASENAME = self.node.process_class._FILE_PRINT_COUNTER_BASENAME
+            else: # xml format
+                print_counter_xml=True
+                self.logger.info('print counter in xml format')
+                FILE_PRINT_COUNTER_BASENAME = self.node.process_class._FILE_XML_PRINT_COUNTER_BASENAME
 
         # Let's pass file handlers to this function
         out_dict, _raw_successful = parse_cp_raw_output(
-            out_folder.open(stdout_filename), out_folder.open(xml_files[0]),
-            out_folder.open(self.node.process_class._FILE_XML_PRINT_COUNTER_BASENAME)
+            out_folder.open(stdout_filename),
+            out_folder.open(xml_files[0]),
+            None if no_trajectory_output else out_folder.open(FILE_PRINT_COUNTER_BASENAME) ,
+            print_counter_xml=print_counter_xml
         )
 
-        # parse the trajectory. Units in Angstrom, picoseconds and eV.
-        # append everthing in the temporary dictionary raw_trajectory
-        raw_trajectory = {}
-        evp_keys = [
-            'electronic_kinetic_energy', 'cell_temperature', 'ionic_temperature', 'scf_total_energy', 'enthalpy',
-            'enthalpy_plus_kinetic', 'energy_constant_motion', 'volume', 'pressure'
-        ]
+        if not no_trajectory_output:
+            # parse the trajectory. Units in Angstrom, picoseconds and eV.
+            # append everthing in the temporary dictionary raw_trajectory
+            raw_trajectory = {}
+            evp_keys = [
+                'electronic_kinetic_energy', 'cell_temperature', 'ionic_temperature',
+                'scf_total_energy', 'enthalpy', 'enthalpy_plus_kinetic',
+                'energy_constant_motion', 'volume', 'pressure'
+            ]
+
+            #order of atom in the output trajectory changed somewhere after 6.5
+            if LooseVersion(out_dict['creator_version']) > LooseVersion('6.5'):
+                new_cp_ordering = True
```

Comment:
Let's see if we can push cp/QE to have a variable in the XML output (or as a head of each file) to specify the format

Reply:
I think we are going to use 6.5 as the last versione where we have the reordering of the atoms, with no new variables in the XML output